src/pydisagg/ihme/splitter/cat_splitter.py

```diff
@@ -0,0 +1,632 @@
+
+    def model_post_init(self, __context: Any) -> None:
+        """
+        Perform extra validation after model initialization.
+
+        Raises
+        ------
+        ValueError
+            If the match criteria in the pattern or population do not match the data.
+        """
+        if not set(self.pattern.index).issubset(self.data.index):
+            raise ValueError(
+                "Match criteria in the pattern must be a subset of the data"
+            )
+        if not set(self.population.index).issubset(
+            self.data.index + self.pattern.index
+        ):
+            raise ValueError(
+                "Match criteria in the population must be a subset of the data and the pattern"
+            )
+        # Check that the 'target' column in pattern and population matches data
+        if self.pattern.target != self.data.target:
+            raise ValueError(
+                "The 'target' column in pattern must match the 'target' column in data"
+            )
+        if self.population.target != self.data.target:
+            raise ValueError(
+                "The 'target' column in population must match the 'target' column in data"
+            )
+
+    def create_ref_return_df(self, data: DataFrame) -> tuple[DataFrame, DataFrame]:
+        """
+        Create reference and return DataFrames.
+
+        Parameters
+        ----------
+        data : DataFrame
+            The input data DataFrame.
+
+        Returns
+        -------
+        tuple[DataFrame, DataFrame]
+            A tuple containing:
+            - ref_df: DataFrame with original data, exploded if necessary.
+            - data: DataFrame with required columns and identifiers.
+        """
+        ref_df = data.copy()
+        ref_df["orig_pyd_id"] = range(
+            len(ref_df)
+        )  # Assign original pyd_id before exploding
+        ref_df["orig_group"] = ref_df[self.data.target]
+        # Explode the 'target' column if it contains lists
+        if ref_df[self.data.target].apply(lambda x: isinstance(x, list)).any():
+            ref_df = ref_df.explode(self.data.target).reset_index(drop=True)
+        # Assign new pyd_id's after exploding
+        ref_df["pyd_id"] = range(len(ref_df))
+        return ref_df, ref_df[self.data.columns + ["pyd_id", "orig_pyd_id"]]
+
+    def parse_data(self, data: DataFrame) -> DataFrame:
+        """
+        Parse and validate the input data DataFrame.
+
+        Parameters
+        ----------
+        data : DataFrame
+            The input data DataFrame.
+
+        Returns
+        -------
+        DataFrame
+            Validated and possibly modified data DataFrame.
+
+        Raises
+        ------
+        KeyError
+            If required columns are missing.
+        ValueError
+            If there are duplicate indices, NaN values, or non-positive values.
+        """
+        name = "While parsing data"
+
+        # Validate core columns first
+        try:
```

Comment:
Why are we working with the `try ... except ...`? If we want better error message, we should directly modify the `validate` functions.